### PR TITLE
feat: exclude empty requests in execution requests list

### PR DESCRIPTION
**Pull request by ensi321**
Related PR ethereum/execution-apis#599

### CODE REVIEW

File: packages/beacon-node/src/execution/engine/types.ts [hunk 1]
```diff
@@ -394,26 +396,39 @@ export function deserializeWithdrawal(serialized: WithdrawalRpc): capella.Withdr
   } as capella.Withdrawal;
 }
 
+function prefixRequests(requestsBytes: Uint8Array, requestType: RequestType): Uint8Array {
+  const prefix = fromHexString(numberToHex(requestType));
```
Comment (nflaig) [inline packages/beacon-node/src/execution/engine/types.ts]: isn't the prefix always one byte?

File: packages/beacon-node/src/execution/engine/types.ts [hunk 2]
```diff
@@ -426,22 +441,64 @@ function deserializeConsolidationRequests(serialized: ConsolidationRequestsRpc):
...
+    return result;
+  }
+
+  let prevRequestType: RequestType | undefined;
+
+  for (const prefixedRequests of serialized) {
+    const currentRequestType = RequestType[prefixedRequests[0] as keyof typeof RequestType];
+    const requests = prefixedRequests.slice(1);
+
+    if (prevRequestType !== undefined && prevRequestType >= currentRequestType) {
```
Comment (nflaig) [inline packages/beacon-node/src/execution/engine/types.ts:487]: why do we have this check, isn't the point of the change to not depend on the order of the requests inside the list?

Comment (ensi321) [reply packages/beacon-node/src/execution/engine/types.ts:487]: > why do we have this check, isn't the point of the change to not depend on the order of the requests inside the list?

There was a debate in the execution api PR: https://github.com/ethereum/execution-apis/pull/599#issuecomment-2505921853

I agree it's pointless to have the ordering requirement, but due to the historical reason, and we want to merge the spec soon, we've decided to keep the ordering requirement. So we need to implement it.

Comment (nflaig) [reply packages/beacon-node/src/execution/engine/types.ts:487]: > but due to the historical reason, and we want to merge the spec soon

~~This seems even more strange, being more strict does not provide any benefit if we don't need it and can parse the values either way due to prefix. It's not clear from the comments to me that being that strict about the order is desired and if so it should be part of the spec.~~

~~I would rather remove the order requirement, we don't gain anything on the CL side from rejecting such a request as we can handle it now with the prefix.~~

nvm this was actually kept from previous spec
> Elements of the list **MUST** be ordered by `request_type` in ascending order

**Review by nflaig — APPROVED** [review]
LGTM

**Comment by philknows** [comment]
Please target devnet-5 branch

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.26.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.26.0) :tada:

**Pull request merged**